src/workerd/jsg/modules-new-test.c++

```diff
@@ -0,0 +1,1889 @@
+    }, [&](Value exception) {
+      js.throwException(kj::mv(exception));
+    });
+  }));
+}
+
+// ======================================================================================
+
+KJ_TEST("Basic types of modules work (text, data, json, wasm)") {
+  PREAMBLE(([&](Lock& js) {
+    ResolveObserver resolveObserver;
+    CompilationObserver compilationObserver;
+    ModuleRegistry::Builder registryBuilder(resolveObserver);
+
+    ModuleBundle::BundleBuilder bundleBuilder;
+    bundleBuilder.addSyntheticModule("abc",
+        Module::newTextModuleHandler(kj::str("hello").releaseArray()));
+    bundleBuilder.addSyntheticModule("xyz",
+        Module::newDataModuleHandler(kj::heapArray<kj::byte>({1,2,3})));
+
+    auto json = kj::str("{\"foo\":123}");
+    bundleBuilder.addSyntheticModule("json",
+        Module::newJsonModuleHandler(json.slice(0, json.size()).attach(kj::mv(json))));
+
+    auto wasm = kj::heapArray<kj::byte>({
+      0x00, 0x61, 0x73, 0x6d, 0x01, 0x00, 0x00, 0x00,
+      0x01, 0x07, 0x01, 0x60, 0x02, 0x7f, 0x7f, 0x01,
+      0x7f, 0x03, 0x02, 0x01, 0x00, 0x07, 0x07, 0x01,
+      0x03, 0x61, 0x64, 0x64, 0x00, 0x00, 0x0a, 0x09,
+      0x01, 0x07, 0x00, 0x20, 0x00, 0x20, 0x01, 0x6a,
+      0x0b,
+    });
+    bundleBuilder.addSyntheticModule("wasm", Module::newWasmModuleHandler(kj::mv(wasm)));
+
+    kj::String mainSource2 =
+        kj::str("export { default as abc } from 'abc';"
+                "export { default as xyz } from 'xyz';"
+                "export { default as json } from 'json';"
+                "export { default as wasm } from 'wasm';"
+                "export { default as wasm2 } from 'wasm?a';");
+
+    bundleBuilder.addEsmModule("worker",
+        mainSource2.slice(0, mainSource2.size()).attach(kj::mv(mainSource2)),
+        Module::Flags::MAIN);
+
+    registryBuilder.add(bundleBuilder.finish());
+
+    auto registry = registryBuilder.finish();
+
+    const auto specifier = "file:///worker"_url;
+
+    ResolveContext resolveContext {
+      .type = ResolveContext::Type::BUNDLE,
+      .source = ResolveContext::Source::OTHER,
+      .specifier = specifier,
+      .referrer = ModuleBundle::BundleBuilder::BASE,
+    };
+    KJ_ASSERT_NONNULL(registry->resolve(resolveContext));
+
+    registry->attachToIsolate(js, compilationObserver);
+
+    js.tryCatch([&] {
+      auto val = ModuleRegistry::resolve(js, "file:///worker", "abc"_kjc);
+      KJ_ASSERT(val.isString());
+      KJ_ASSERT(kj::str(val) == "hello"_kjc);
+    }, [&](Value exception) {
+      js.throwException(kj::mv(exception));
+    });
+
+    js.tryCatch([&] {
+      auto val = ModuleRegistry::resolve(js, "file:///worker", "xyz"_kjc);
+      KJ_ASSERT(val.isArrayBuffer());
+    }, [&](Value exception) {
+      js.throwException(kj::mv(exception));
+    });
+
+    js.tryCatch([&] {
+      auto val1 = ModuleRegistry::resolve(js, "file:///worker", "json"_kjc);
+      auto val2 = ModuleRegistry::resolve(js, "file:///json", "default"_kjc);
```

Comment:
Nit: the default value for `exportName` already is "default", it can be omitted here.

Reply:
I know, I just prefer to be more explicit